mkosi.clangd

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+set -eux
+
+SUBDIR="$1"
+shift
+
+if [[ -n "${MKOSI_CONFIG-}" ]]; then
+    exec mkosi-chroot env --chdir="$SRCDIR/$SUBDIR" clangd "$@"
+fi
+
+MKOSI_CONFIG="$(mkosi --directory "$(dirname "$(readlink -f "$0")")" --json summary | jq -r .Images[-1])"
+
+DISTRIBUTION="$(jq -r .Distribution <<< "$MKOSI_CONFIG")"
+RELEASE="$(jq -r .Release <<< "$MKOSI_CONFIG")"
+ARCH="$(jq -r .Architecture <<< "$MKOSI_CONFIG")"
+
+BUILDDIR="$(jq -r .BuildDirectory <<< "$MKOSI_CONFIG")"
+CACHEDIR="$(jq -r .CacheDirectory <<< "$MKOSI_CONFIG")"
+
+exec mkosi \
+    --directory "$(dirname "$(readlink -f "$0")")" \
+    --incremental=strict \
+    --build-sources-ephemeral=no \
+    --format=none \
+    --build-script= \
+    --build-script="$(readlink -f "$0")" \
+    build \
+    "$SUBDIR" \
+    --enable-config \
+    --compile-commands-dir=/work/build/"$SUBDIR" \
+    --path-mappings="\
+$(pwd)=/work/src/$SUBDIR,\
```

Comment:
I do not think that this works when using out-of-tree modules or anything else which uses more than one `BuildSources=` which is why I went for the more complicated `jq` logic

Reply:
@septatrix I'm not sure I want to accomodate out-of-tree modules. I think if you're using this at all you're a strong proponent of open source and should be willing to develop modules in tree, not out-of-tree.

Reply:
Sure, in-tree is preferred over out-of-tree but I see no reason to hinder their DX if they want to use this tool, and there might also be cases where people compose their build sources from multiple places (though that is probably rather rare).

Finally, this again depends on `$PWD` as opposed to just reading the from the json summary which would be much more stable and robust. I see no disadvantage is using the more general variant here...